quizblock/models.py

```diff
@@ -473,24 +470,25 @@ def metadata(self):
         return row
 
     def user_value(self, user):
-        value = ''
-        r = self._submission_cache.filter(user=user).order_by("-submitted")
-        if r.count() == 0:
-            # user has not answered this question
-            return None
-        submission = r.first()
-        r = self._response_cache.filter(submission=submission)
-        if r.count() > 0:
-            if (self.question.is_short_text() or
-                    self.question.is_long_text()):
-                value = r.first().value
-            elif self.question.is_multiple_choice():
-                if self.answer.value in [res.value for res in r]:
-                    value = self.answer.id
-            else:  # single choice
-                for a in self._answer_cache:
-                    if a.value == r.first().value:
-                        value = a.id
-                        break
-
-        return smart_str(value)
+        value = None
+        submission = Submission.objects.filter(
+            quiz=self.question.quiz, user=user).order_by("-submitted").first()
```

Comment:
I'm just curious, is there a reason you decided not to use `self._submission_cache` here?